test/unit/bson_regex.test.js

```diff
@@ -0,0 +1,43 @@
+'use strict';
+
+const expect = require('chai').expect;
+const BSONRegExp = require('bson').BSONRegExp;
+
+describe('BSONRegExp', () => {
+  describe('bsonRegExp option', () => {
+    // define option for tests to use
+    const option = { bsonRegExp: true };
+    for (const passOptionTo of ['client', 'db', 'collection', 'operation']) {
+      it(`should respond with BSONRegExp class with option passed to ${passOptionTo}`, function(done) {
+        const client = this.configuration.newClient(passOptionTo === 'client' ? option : undefined);
+        let collection;
+
+        return client
```

Comment:
since we are using done, this `return` should be removed, otherwise mocha gets confused